inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/cls/ClsResourceOperator.java

```diff
@@ -136,14 +169,34 @@ private ClsClient getClsClient(ClsDataNodeDTO clsDataNode) {
     /**
      * Create topic index by tokenizer
      */
-    private void createTopicIndex(SinkInfo sinkInfo) throws BusinessException {
+    private void createTopicIndex(SinkInfo sinkInfo, ClsDataNodeDTO clsDataNode) throws BusinessException {
+
         ClsSinkDTO clsSinkDTO = JsonUtils.parseObject(sinkInfo.getExtParams(), ClsSinkDTO.class);
-        if (StringUtils.isNotBlank(clsSinkDTO.getTokenizer())) {
-            LOG.warn("topic {} tokenizer is empty", clsSinkDTO.getTopicName());
+        if (StringUtils.isBlank(clsSinkDTO.getTokenizer())) {
+            LOG.warn("topic {} tokenizer is blank", clsSinkDTO.getTopicName());
+            return;
+        }
+        FullTextInfo topicIndexFullText = getTopicIndexFullText(sinkInfo, clsDataNode);
+        if (ObjectUtils.anyNotNull(topicIndexFullText)) {
+            // if topic index exist,update
+            updateTopicIndex(sinkInfo, clsDataNode);
             return;
         }
-        ClsDataNodeDTO clsDataNode = getClsDataNode(sinkInfo);
         ClsClient clsClient = getClsClient(clsDataNode);
+        CreateIndexRequest req = getCreateIndexRequest(clsSinkDTO);
+        try {
+            clsClient.CreateIndex(req);
```

Comment:
Please add log for `CreateIndexResponse`.

Reply:
done